go/gilded-rose_test.go

```diff
@@ -1,15 +1,151 @@
 package main
 
-import "testing"
-
+import (
+	"fmt"
+	"testing"
+	"strconv"
+)
+func Print_Items(items []*Item){
+	fmt.Println("name, sellIn, quality")
+	for i := 0; i < len(items); i++ {
+		fmt.Println(items[i])
+	}
+	fmt.Println("")
+}
 func Test_Foo(t *testing.T) {
 	var items = []*Item{
-		&Item{"foo", 0, 0},
+		&Item{"+5 Dexterity Vest", 10, 20},
+		&Item{"Aged Brie", 2, 0},
+		&Item{"Elixir of the Mongoose", 5, 7},
+		&Item{"Sulfuras, Hand of Ragnaros", 0, 80},
+		&Item{"Sulfuras, Hand of Ragnaros", -1, 80},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 15, 20},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 10, 49},
+		&Item{"Backstage passes to a TAFKAL80ETC concert", 5, 49},
+		&Item{"Conjured Mana Cake", 15, 40}, // <-- :O
+		&Item{"Conjured Mana Cake", 3, 46}, // <-- :O
+	}
+
+	days := 12
+
+	for day := 0; day < days; day++ {
+		fmt.Printf("-------- day %d --------\n", day)
+		Print_Items(items)
+		UpdateQuality(items)
+	}
+	fmt.Printf("-------- LAST day %d --------\n", days)
+	Print_Items(items)
+
+	//+5 Dexterity Vest
+	if items[0].name != "+5 Dexterity Vest" {
+		t.Errorf("Name: Expected %s but got %s ", "+5 Dexterity Vest", items[0].name)
+	}
+	if items[0].sellIn != -2 {
+		t.Errorf("Sellin: Expected %s but got %s ", "-2", strconv.Itoa(items[0].sellIn))
+	}
+	if items[0].quality != 6 {
+		t.Errorf("Quality: Expected %s but got %s ", "6", strconv.Itoa(items[0].quality))
+	}
```

Comment:
In Go it's considered idiomatic to use tables to build your tests, by tables they actually mean a slice of structs. This way you can build a tree/table of test cases, the expected results, a name or description of what it's testing, and then have a very simple func that iterates over those and executes the tests.

The beauty in doing that is that you can develop comprehensive sets of test cases, but they're all quite uniform, well structured, and easy to navigate/read, and debug. You can see my poor example of this for this challenge here... https://github.com/ContainerSolutions/GildedRose-Refactoring-Kata/pull/4/files#diff-e69fd5910b4361c4910993535824102893a501bfb31d1dbdfd696d7f3ea28a31R12-R193